core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -989,8 +989,8 @@ private[spark] class TaskSetManager(
         if (ef.className == classOf[TaskOutputFileAlreadyExistException].getName) {
           // If we can not write to output file in the task, there's no point in trying to
           // re-execute it.
-          logError(log"Task ${MDC(TASK_ID, info.id)} in stage ${MDC(STAGE_ID, taskSet.id)} " +
-            log"(TID ${MDC(TID, tid)}) can not write to output file: " +
+          logError(log"Task ${MDC(TASK_INFO_ID, info.id)} in stage ${MDC(STAGE_ID, taskSet.id)} " +
```

Comment:
the `taskSet.id` is `s"$stageId.$stageAttemptId"`

Reply:
for cases like TASK_INFO_ID and TASK_SET_ID, I wondering if we can just expose

- TASK_INDEX
- TASK_ATTEMP_NUM
- STAGE_ID
- STAGE_ATTEMP_ID

to the MDC?

Reply:
Good point, updated.